File: lib/domain/pesticide/dtos/pesticide_application_detail_snapshot.rb
# frozen_string_literal: true

module Domain
  module Pesticide
    module Dtos
      class PesticideApplicationDetailSnapshot
        attr_reader :dilution_ratio, :amount_per_m2, :amount_unit, :application_method

        def initialize(dilution_ratio:, amount_per_m2:, amount_unit:, application_method:)
          @dilution_ratio = dilution_ratio
          @amount_per_m2 = amount_per_m2
          @amount_unit = amount_unit
          @application_method = application_method
        end
      end
    end
  end
end

File: crates/agrr-domain/src/pesticide/dtos/pesticide_application_detail_snapshot.rs
/// Ruby: `Domain::Pesticide::Dtos::PesticideApplicationDetailSnapshot`
#[derive(Debug, Clone, PartialEq)]
pub struct PesticideApplicationDetailSnapshot {
    pub dilution_ratio: Option<String>,
    pub amount_per_m2: Option<f64>,
    pub amount_unit: Option<String>,
    pub application_method: Option<String>,
}
